### ISSUE TITLE
Finetune on 3090 but loss equal to zero

### ISSUE THREAD

**Issue by Yingyue-L** [havenhq/mamba-chat#11]
Thanks for the great work! I tried to finetune the Mamba model using four 3090 GPUs following your code. But there's a problem: the loss drops to zero after just two steps. Can you help me figure out what's going wrong and fix it? 
![image](https://github.com/havenhq/mamba-chat/assets/56422335/9d87c4d5-7574-492d-8896-e9d5632a616e)

**Comment by justusmattern27** [comment]
Interesting - have you modified `train_mamba.py` in some way?

**Comment by Yingyue-L** [comment]
I have only modified the `model path` to the `local_dir`, everything else remained unchanged.

**Comment by Zong-Bin** [comment]
I encountered the same problem.

**Comment by DifferentialityDevelopment** [comment]
Figured out the problem, I was busy fine tuning the mamba-370m since I only have a measly 2070 super with 8GB VRAM, finally realised that what was happening was that the model dimensions didn't match up with the max_tokens set in ChatDataModule, once I lowered it to 1024 my logits stopped giving NaN and the loss was calculated correctly on subsequent steps.

in train_mamba.py, change the data_module to this
`
data_module = ChatDataModule(
        tokenizer=tokenizer,
        data_path=args.data_path,
        conversation_template=tokenizer.chat_template,
        max_tokens=model.config.d_model # <- reads the model dimension directly from the config.
    )
`

**Comment by DifferentialityDevelopment** [comment]
Okay so I think I've now finally gotten down to the issue, I had to switch to float16 to bfloat16 wouldn't work on my 2070 super, that's what was causing my logits to become NaN, switching to float32 made the issue go away and learning rate no longer suddenly drop to 0.
![image](https://github.com/havenhq/mamba-chat/assets/9594229/cc79e473-ca40-4455-b9b9-7d3046717808)
